torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -240,6 +245,11 @@ def check_cacheable(gm: torch.fx.GraphModule):
     for node in nodes:
         check_node_safe(node)
 
+    for n, m in gm.named_modules():
+        # Check all submodules?
+        if "saved_tensors_hooks" in n:
+            check_cacheable(m)
```

Comment:
correct me if I'm wrong @IvanKobzarev - I think the reason you have this extra loop logic is because these saved_tensor_hooks are a bit special: they show up as subgraphs on the dynamo `GraphModule`, that are **not** used anywhere in the top-level graph.

Contrast that to most HOPs, which show up as subgraphs, but are called from somewhere in the top level graph. (fyi @jamesjwu )

Given that - maybe instead of looping throguh all `gm.named_moodules`, we should just check for the existence of  these specific `saved_tensor_hooks` submods? `if hasattr(gm, 'saved_tensor_hooks_pack_graph')`, etc

Reply:
The exact name of the subgraph is known only in `_dynamo/output_graph.py` after calling `install_subgraph()`.
Atm it is called `install_subgraph(gm, "saved_tensors_hooks_pack") and installs "saved_tensors_hooks_pack_0".
Technically we could hardcode "saved_tensors_hooks_pack_0" and "saved_tensors_hooks_unpack_0" here and assert in output_graph that this was intalled under this name. With this we can skip looping.